src/main/java/org/folio/rest/impl/UserGroupAPI.java

```diff
@@ -40,323 +32,176 @@
-          new String[]{"*"}, cql, true, true,
-            reply -> {
-              try {
-                if(reply.succeeded()){
-                  Usergroups groups = new Usergroups();
-                  @SuppressWarnings("unchecked")
-                  List<Usergroup> groupList = reply.result().getResults();
-                  groups.setUsergroups(groupList);
-                  groups.setTotalRecords(reply.result().getResultInfo().getTotalRecords());
-                  asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(GetGroupsResponse
-                    .respond200WithApplicationJson(groups)));
-                }
-                else{
-                  log.error(reply.cause().getMessage(), reply.cause());
-                  asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(GetGroupsResponse
-                    .respond400WithTextPlain(reply.cause().getMessage())));
-                }
-              } catch (Exception e) {
-                log.error(e.getMessage(), e);
-                asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(GetGroupsResponse
-                  .respond500WithTextPlain(messages.getMessage(
-                    lang, MessageConsts.InternalServerError))));
-              }
-            });
-      }
-      catch(FieldException fe){
-        log.error(fe.getLocalizedMessage(), fe);
-        asyncResultHandler.handle(Future.succeededFuture(GetUsersResponse.respond400WithTextPlain(
-                "CQL Parsing Error for '" + query + "': " + fe.getLocalizedMessage())));
-      }
-      catch (Exception e) {
-        log.error(e.getMessage(), e);
-        asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(GetGroupsResponse
-          .respond500WithTextPlain(messages.getMessage(lang, MessageConsts.InternalServerError))));
-      }
-    });
+    String lang, Map<String, String> okapiHeaders,
+    Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
 
+    PgUtil.get(GROUP_TABLE, Usergroup.class, Usergroups.class, query, offset, limit, okapiHeaders,
+      vertxContext, GetGroupsResponse.class, asyncResultHandler);
   }
 
   @Validate
   @Override
   public void postGroups(String lang, Usergroup entity,
-          Map<String, String> okapiHeaders,
-          Handler<AsyncResult<Response>> asyncResultHandler,
-          Context vertxContext) {
-
-    vertxContext.runOnContext(v -> {
-      try {
-        System.out.println("sending... postGroups");
-        String id = entity.getId();
-        if(id == null) {
-          id = UUID.randomUUID().toString();
-          entity.setId(id);
+    Map<String, String> okapiHeaders,
+    Handler<AsyncResult<Response>> asyncResultHandler,
+    Context vertxContext) {
+
+    String id = entity.getId();
+    if (id == null) {
+      id = UUID.randomUUID().toString();
+      entity.setId(id);
+    }
+    PostgresClientUtil.getInstance(vertxContext, okapiHeaders).save(
```

Comment:
Consider using PgUtil.post

Reply:
That results in a regression.
Failed tests: 
  RestVerticleIT.test2Group:1737 Test failed

Reply:
This is caused by the "try to add a duplicate group" test.
PgUtil.post correctly returns 400 with a plain/text body. This complies with https://github.com/folio-org/raml/blob/raml1.0/rtypes/collection.raml#L38-L70

The RestVerticleIT.test2Group unit test expects a 422 application/json body:
https://github.com/folio-org/raml/blob/raml1.0/traits/validation.raml
The validation trait is optional for any post endpoint and is intended for additional tests that goes beyond what the regular collection.raml checks.

If you think that PgUtil.post should check for the optional 422 validation trait please file an issue against PgUtil.

Otherwise consider to either accept any return code (400 or 422), or remove the validation trait.